### ISSUE TITLE
ImagePicker Crash if quality != 1

### ISSUE THREAD

**Issue by LipLoody** [expo/expo#15119]
### Summary

When using hermes, expo-dev-client and ImagePicker (launchImageLibraryAsync or getCameraPermissionsAsync) the app crash after picking an image.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

bare

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

43

### Environment

  Expo CLI 4.12.12 environment info:
    System:
      OS: Linux 5.11 Ubuntu 20.04.3 LTS (Focal Fossa)
      Shell: 5.8 - /usr/bin/zsh
    Binaries:
      Node: 14.18.1 - /usr/bin/node
      Yarn: 1.22.10 - /usr/bin/yarn
      npm: 8.1.3 - /usr/bin/npm
    npmPackages:
      @expo/webpack-config: ^0.16.2 => 0.16.4 
      expo: ^43.0.0 => 43.0.0 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.2 => 0.64.2 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      expo-cli: 4.12.12
    Expo Workflow: bare

### Reproducible demo or steps to reproduce from a blank project

Create project with `"jsEngine": "hermes",` in app.json 

add this code : 
```
let result = await ImagePicker.launchImageLibraryAsync({ // same with : "getCameraPermissionsAsync"
        mediaTypes: ImagePicker.MediaTypeOptions.Images,
        allowsEditing: false,
        quality: 0.5,
        base64: true
      });
```

run the app in expo-dev-client
```
expo run:android
```

After picking an image, you can see in logcat a crash like:

```
11-09 19:20:46.970 26256 26256 I AwareBitmapCacher: handleReleaseCache: pid=26256
11-09 19:20:46.972 26256 26256 I AwareBitmapCacher: init lrucache size: 2097152 pid=26256
11-09 19:20:47.044 26256 26256 D AndroidRuntime: Shutting down VM
--------- beginning of crash
11-09 19:20:47.045 26256 26256 E AndroidRuntime: FATAL EXCEPTION: main
11-09 19:20:47.045 26256 26256 E AndroidRuntime: Process: com.example.app, PID: 26256
11-09 19:20:47.045 26256 26256 E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=2, result=-1, data=Intent { dat=content://com.android.providers.media.documents/document/image:140514 flg=0x1 }} to activity {com.example.app/com.example.app.MainActivity}: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter mImageLoader
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4936)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4978)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:49)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2123)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:109)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:207)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7470)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:524)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:958)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: Caused by: java.lang.NullPointerException: Parameter specified as non-null is null: method kotlin.jvm.internal.Intrinsics.checkNotNullParameter, parameter mImageLoader
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at expo.modules.imagepicker.exporters.CompressionImageExporter.<init>(Unknown Source:2)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at expo.modules.imagepicker.ImagePickerModule.handleOnActivityResult(ImagePickerModule.kt:396)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at expo.modules.imagepicker.ImagePickerModule.onActivityResult(ImagePickerModule.kt:314)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at expo.modules.adapters.react.services.UIManagerModuleWrapper$4.onActivityResult(UIManagerModuleWrapper.java:165)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at com.facebook.react.bridge.ReactContext.onActivityResult(ReactContext.java:305)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at com.facebook.react.ReactInstanceManager.onActivityResult(ReactInstanceManager.java:762)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at com.facebook.react.ReactDelegate.onActivityResult(ReactDelegate.java:90)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at com.facebook.react.ReactActivityDelegate.onActivityResult(ReactActivityDelegate.java:112)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at expo.modules.ReactActivityDelegateWrapper.onActivityResult(ReactActivityDelegateWrapper.kt:80)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at com.facebook.react.ReactActivity.onActivityResult(ReactActivity.java:68)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:7775)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4929)
11-09 19:20:47.045 26256 26256 E AndroidRuntime: 	... 11 more
11-09 19:20:47.053 26256 26256 I Process : Sending signal. PID: 26256 SIG: 9
```

**Comment by iainscanlon** [comment]
Can confirm this is also happening to me

**Comment by mlecoq** [comment]
I have the same issue - I don't use `"jsEngine": "hermes"`

**Comment by WilliamAlexander** [comment]
same issue here. 
If I set allowsEditing : true and quality .. .say 0.5. It's all fine
If I set allowsEditing : false and quality to anything but 1...then crash

**Comment by Pazago** [comment]
Same issue for me.
In Expo Go everything is fine.
When I make a EAS build for an APK it crashes with same logcat error like LipLoody on a hardware android device and on a virtual device. Same for expo run:android

If I change the quality back to 1 and allowsEditing: false it works on all devices

**Comment by WilliamAlexander** [comment]
I have the solution.

In expo-image-picker, there's a Kotlin file called ImagePickerModule.kt.

It has the following logic
```
val exporter: ImageExporter = if (pickerOptions.quality == ImagePickerConstants.DEFAULT_QUALITY) {
    RawImageExporter(contentResolver, pickerOptions.isBase64)
  } else {
    CompressionImageExporter(mImageLoader, pickerOptions.quality, pickerOptions.isBase64)
  }
```
      
 When we set quality to anything but 1 the following line is run
 ```
 CompressionImageExporter(mImageLoader, pickerOptions.quality, pickerOptions.isBase64)
 ```
 
The problem is mImageLoader is null.
The reason for this is **ImageLoaderInterface** cannot be found.
To fix that you need to 

```
npm install expo-image-loader
```
expo-image-loader has that interface.

Not sure why this was not documented, maybe I missed it :)

I tested it on as follows and it works fine. Tested in 2 physical Android devices

```
let result = await ImagePicker.launchCameraAsync({
    mediaTypes: ImagePicker.MediaTypeOptions.Images,
    allowsEditing: false,
    quality: 0.1,
    base64: true,
    allowsMultipleSelection: false
});

```


If this helps people, please do something kind this Xmas. 
If you see a homeless person, toss them a coin and a smile
xx

**Comment by Lastofthefirst** [comment]
Good Lord! I've been struggling to figure this out for a year. That worked. Thank you.

**Comment by zeulb** [comment]
This is also an issue with us on Android with Hermes engine, installing `expo-image-loader` fixed it, but i don't see that documented anywhere

**Comment by superjose** [comment]
OMG, WilliamAlexander's did the trick! I just had to do `eas build` again and update the app on my device and it did the trick!

**Comment by kierrapalmer** [comment]
Adding `expo-image-loader ` caused all sorts of errors for me when recompiling. I had to run `expo doctor --fix-dependencies` to get it to be able to build again.

**Comment by barthap** [comment]
Could anybody check if it's still an issue for you in [SDK 46 beta](https://blog.expo.dev/expo-sdk-46-beta-is-now-available-9dfee4040aa7)?

**Comment by emeraldsanto** [comment]
@barthap I could reproduce something that seems to fit this problem, though I don't have the logs to prove it, on Expo SDK `46.0.15` in a bare React Native application. I will try to ship the above fix and check in with affected users.
